static/js/sentry.ts

```diff
@@ -0,0 +1,48 @@
+import * as Sentry from "@sentry/browser";
+import {HttpClient as HttpClientIntegration} from "@sentry/integrations";
+import {BrowserTracing} from "@sentry/tracing";
+import _ from "lodash";
+
+import {page_params} from "./page_params";
+
+if (page_params.server_sentry_dsn) {
+    const url_regex = new RegExp("^" + _.escapeRegExp(page_params.realm_uri) + "/");
+    Sentry.init({
+        dsn: page_params.server_sentry_dsn,
+        environment: page_params.development_environment ? "development" : "production",
+
+        release: "zulip-server@" + ZULIP_VERSION,
+        integrations: [
+            new BrowserTracing({
+                tracePropagationTargets: [url_regex],
+            }),
+            new HttpClientIntegration({
+                failedRequestStatusCodes: [500, 502, 503, 504],
+                failedRequestTargets: [url_regex],
+            }),
+        ],
+        allowUrls: [url_regex, page_params.webpack_public_path],
+        sampleRate: 1,
+        tracesSampleRate: 1,
+        initialScope: {
```

Comment:
So one thing that I think will be very helpful is the flag of "Are we inside an unofficial Electron app"? Ferdi and similar third-party Electron apps send a lot of error reporting spam that we can't ourselves fix, and probably will want to filter in some way in our default view. I don't know if there is anything we can/should do to try to tag that here vs. just in the Sentry UI, but wanted to flag that as thing to pay attention to.

Reply:
Understood that that's a priority to expose.  However, I don't see it exposing that information via its user-agent, so we'll need to do some spelunking to figure out how to detect it, which seems like a follow-up PR.

Reply:
Sounds good!